File: src/application/use_cases/contexts.py
"""Context-related application use cases."""

from __future__ import annotations

from src.application.ports import ContextSwitcher
from src.domain.models import OperationError


def set_current_context(
    context_name: str,
    switcher: ContextSwitcher,
    *,
    require_confirmation: bool,
    confirmed: bool,
) -> OperationError | None:
    if require_confirmation and not confirmed:
        return OperationError(
            code="confirmation_required",
            message="Context switch requires explicit confirmation",
        )

    return switcher.switch_context(context_name)

File: internal/application/usecases/contexts.go
package usecases

import (
	"github.com/systemframe/k3ctx/internal/domain"
)

// SetCurrentContext switches the active kubectl context.
// When requireConfirmation is true and confirmed is false, returns a confirmation_required error.
func SetCurrentContext(contextName string, switcher domain.ContextSwitcher, requireConfirmation, confirmed bool) *domain.OperationError {
	if requireConfirmation && !confirmed {
		err := domain.OperationError{
			Code:    "confirmation_required",
			Message: "User confirmation required to switch context",
		}
		return &err
	}
	return switcher.SwitchContext(contextName)
}
